src/WordPressCodeExtractor.php

```diff
@@ -109,21 +111,37 @@ public static function fromDirectory( $dir, Translations $translations, array $o
 	 * Recursively gets all PHP files within a directory.
 	 *
 	 * @param string $dir A path of a directory.
+	 * @param array $exclude List of files and directories to skip.
 	 *
 	 * @return array File list.
 	 */
-	protected static function getFilesFromDirectory( $dir ) {
+	protected static function getFilesFromDirectory( $dir, array $exclude = [] ) {
 		$filtered_files = [];
 
 		try {
 			$files = new RecursiveIteratorIterator(
-				new RecursiveDirectoryIterator( $dir, RecursiveDirectoryIterator::SKIP_DOTS ),
+				new RecursiveCallbackFilterIterator(
+					new RecursiveDirectoryIterator( $dir, RecursiveDirectoryIterator::SKIP_DOTS ),
+					function ( $file, $key, $iterator ) use ( $exclude ) {
+						/** @var SplFileInfo $file */
+						if ( in_array( $file->getBasename(), $exclude, true ) ) {
```

Comment:
This will fail if a path is returned with a trailing slash. Using trailing slashes is often done for avoiding partial matches. This either needs specific handling of trailing slashes, or an explicit mention in the argument documentation.

Reply:
I don't think `getBasename()` ever returns a path with a trailing slash. The values in `$exclude` could have indeed have trailing slashes, yes.

I guess I can safely add a change to remove any slashes at the beginning and the end of the excluded directories/files and document this.